### PR TITLE
send the Content-Type header when uploading blobs

### PR DESCRIPTION
**Pull request by paulo-santos**
This change sets the correct content-type when uploading files. The script was setting "application/octet-stream" for all files.

### CODE REVIEW

**Comment by paulo-santos** [comment]
The build failed only for pypy3, but I can't figure out why.
Everything is green when running in travis from my fork https://travis-ci.org/paulo-santos/azure-batch-samples/builds/83088173

File: Python/Storage/blobxfer.py [hunk 1]
```diff
@@ -67,6 +67,8 @@
 import errno
 import hashlib
 import os
+import mimetypes
```
Comment (alfpark) [inline Python/Storage/blobxfer.py]: Place in alphabetical order

File: Python/Storage/blobxfer.py [hunk 2]
```diff
@@ -304,7 +306,7 @@ def get_blob_properties(self, container_name, blob_name):
 
     def put_blob(
             self, container_name, blob_name, blob, x_ms_blob_type,
-            x_ms_blob_content_md5, x_ms_blob_content_length):
+            x_ms_blob_content_md5, x_ms_blob_content_length, x_ms_blob_content_type="application/octet-stream"):
```
Comment (alfpark) [inline Python/Storage/blobxfer.py]: Wrong parameter order, this must match the azure storage sdk ordering (place before md5). Also, do not have a default of a string, default should be None.

Additionally make sure your code is flake8 clean.

File: Python/Storage/blobxfer.py [hunk 3]
```diff
@@ -401,7 +405,7 @@ def put_block(
 
     def put_block_list(
             self, container_name, blob_name, block_list,
-            x_ms_blob_content_md5):
+            x_ms_blob_content_md5, x_ms_blob_content_type='application/octet-stream'):
```
Comment (alfpark) [inline Python/Storage/blobxfer.py]: Same issue here; place before md5, remove default.

File: Python/Storage/blobxfer.py [hunk 4]
```diff
@@ -724,6 +731,9 @@ def compute_md5_for_file_asbase64(filename, pagealign=False, blocksize=65536):
         else:
             return str(base64.b64encode(hasher.digest()), 'ascii')
 
+def get_mime_type(filename):
```
Comment (alfpark) [inline Python/Storage/blobxfer.py]: Add docstring, follow format of other functions

File: Python/Storage/blobxfer.py [hunk 5]
```diff
@@ -724,6 +731,9 @@ def compute_md5_for_file_asbase64(filename, pagealign=False, blocksize=65536):
...
+    mimetype = ( mimetypes.guess_type(filename)[0] or'application/octet-stream')
```
Comment (alfpark) [inline Python/Storage/blobxfer.py]: combine as: return mimetypes.guess_type(...

File: Python/Storage/blobxfer.py [hunk 6]
```diff
@@ -724,6 +731,9 @@ def compute_md5_for_file_asbase64(filename, pagealign=False, blocksize=65536):
...
+    return mimetype
```
Comment (alfpark) [inline Python/Storage/blobxfer.py]: change to: return mimetypes.guess_type(filename)[0] or ...

**Comment by paulo-santos** [comment]
Hey @alfpark, changes suggested by your code review implemented. Please take a look.

**Comment by alfpark** [comment]
Thanks @paulo-santos for addressing the comments. Can you please perform a squash, then I'll merge the PR.

**Comment by paulo-santos** [comment]
@alfpark Commits have been squashed! :)

**Pull request merged**